src/collision/collider_backend.rs

```diff
@@ -372,13 +371,28 @@ fn update_aabb<C: AnyCollider>(
         (&Position, Option<&LinearVelocity>, Option<&AngularVelocity>),
         With<Children>,
     >,
-    dt: Res<Time>,
-    narrow_phase_config: Option<Res<NarrowPhaseConfig>>,
+    narow_phase_config: Res<NarrowPhaseConfig>,
+    length_unit: Res<PhysicsLengthUnit>,
+    time: Res<Time>,
 ) {
-    // Safety margin multiplier bigger than DELTA_TIME to account for sudden accelerations
-    let safety_margin_factor = 2.0 * dt.delta_seconds_adjusted();
+    let delta_secs = time.delta_seconds_adjusted();
+    let default_speculative_margin = length_unit.0 * narow_phase_config.default_speculative_margin;
+    let contact_tolerance = length_unit.0 * narow_phase_config.contact_tolerance;
+
+    for (collider, mut aabb, pos, rot, collider_parent, speculative_margin, lin_vel, ang_vel) in
```

Comment:
This seems like a good place to iterate in parallel using `for_each`

Reply:
I got mixed results when running the benchmarks, same for the other `for_each` suggestions. Overall it seems like it just causes performance regressions due to the added overhead, especially for smaller scenes.

With a scene that has a lot more physics objects but less active collisions, it might be more worth it, but I think we would need to trace a more realistic large-scale game scene to be certain, and we don't really have that currently. So I think I'll leave it for now, but it might be worth looking into and benchmarking further in the future.